noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_circuit_public_inputs_composer.nr

```diff
@@ -186,12 +188,29 @@ impl PrivateKernelCircuitPublicInputsComposer {
     fn propagate_logs(&mut self, source: DataSource) {
         self.public_inputs.end.encrypted_logs_hashes.extend_from_bounded_vec(array_to_bounded_vec(source.private_call_public_inputs.encrypted_logs_hashes));
         self.public_inputs.end.unencrypted_logs_hashes.extend_from_bounded_vec(array_to_bounded_vec(source.private_call_public_inputs.unencrypted_logs_hashes));
+        let note_logs = source.private_call_public_inputs.note_encrypted_logs_hashes;
+        for i in 0..note_logs.len() {
+            if !is_empty(note_logs[i]) {
+                let note_index = self.match_log_to_note(note_logs[i]);
+                assert(
+                    note_index != MAX_NEW_NOTE_HASHES_PER_TX, "Could not find note hash linked to note log."
```

Comment:
```suggestion
                    note_log.note_hash_counter == self.public_inputs.end.new_note_hashes.get(note_index).counter(), "Could not find note hash linked to note log."
```

Reply:
It does - and still does so even with `&`. I can add the check above after the max check? So the whole statement is:
```
            if !is_empty(note_logs[i]) {
                let note_index = self.match_log_to_note(note_logs[i]);
                assert(
                    note_index != MAX_NEW_NOTE_HASHES_PER_TX, "Could not find note hash linked to note log."
                );
                assert(
                    note_logs[i].note_hash_counter == self.public_inputs.end.new_note_hashes.get(note_index).counter(), "Could not find note hash linked to note log."
                );
                self.public_inputs.end.note_encrypted_logs_hashes.push(note_logs[i]);
            }
```

Reply:
It's fine as long as it fails when it couldn't find a note hash that has the same counter. But if we want to make it throw a clearer message when the index is invalid, maybe change the first line to `note_index < MAX_NEW_NOTE_HASHES_PER_TX`.
And the second line can use `get_unchecked`.